parties/googleclient.go

```diff
@@ -64,15 +66,19 @@ func (g *GoogleOp) RequestTokens(cicHash string, cb TokenCallback) error {
 		return uuid.New().String()
 	}
 
+	ch := make(chan []byte)
+	chErr := make(chan error)
+
 	http.Handle("/login", rp.AuthURLHandler(state, provider, rp.WithURLParam("nonce", cicHash)))
 
 	marshalToken := func(w http.ResponseWriter, r *http.Request, tokens *oidc.Tokens[*oidc.IDTokenClaims], state string, rp rp.RelyingParty) {
 		if err != nil {
 			http.Error(w, err.Error(), http.StatusInternalServerError)
+			chErr <- err
 			return
 		}
-		cb(tokens)
 
+		ch <- []byte(tokens.IDToken)
```

Comment:
Using channels rather than a callback here has been on my mental task list for longer than I'd like to admit. Thank you for doing this, so much easier to read and think about